### ISSUE TITLE
Random flickering after trying to link with homeassistant

### ISSUE THREAD

**Issue by chaayac** [albertogeniola/meross-homeassistant#375]
**Describe the bug**
At seemingly random intervals, all my bulbs flicker briefly but at the same time. You can check this video for what it looks like:

https://drive.google.com/file/d/1MCFcoi5i17xx9fj1xUcPSD2rzh5Z_Fdf/view?usp=sharing

The bottom 2 bulbs were the old bulbs, the top 2 were new bulbs that had not yet been linked to my meross account.

Disclaimer: I actually have no idea if the link to HA was causing this issue, only that it started occuring after linking my msl120d bulbs to homeassistant. I have since unlinked them from HA and actually shut down my HA server to be sure. I also reset the bulbs and re-linked them to my meross account and Google Home. I contacted Meross support and they sent me new bulbs to try this with, but it even happens with the new bulbs after linking to my meross account.

**Your environment**
Hardware environment: msl120d Firmware version 2.1.2

**Logs taken when the issue happened**
I used the sniffer tool to sniff an affected device, and these were the logs from the sniffer when the flicker occurred. Can someone help me make sense of them?

```
2022-07-12 14:54:31,868 - Sniffer - DEBUG - Initializing the MQTT connection...
2022-07-12 14:54:32,389 - Sniffer - DEBUG - (Re)Starting the MQTT loop.
2022-07-12 14:54:32,565 - Sniffer - DEBUG - Connected to MQTT Broker
2022-07-12 14:54:32,566 - Sniffer - INFO - Subscribing to topic: /appliance/21020594730951290d4248e1e94ae23a/subscribe
2022-07-12 14:54:32,735 - Sniffer - DEBUG - Subscribed correctly
2022-07-12 14:54:32,735 - Sniffer - INFO - Subscribing to topic: /app/1583979-sniffer/subscribe
2022-07-12 14:54:32,934 - Sniffer - DEBUG - Subscribed correctly
2022-07-12 14:54:32,935 - Sniffer - INFO - Subscribing to topic: /app/1583979/subscribe
2022-07-12 14:54:33,106 - Sniffer - DEBUG - Subscribed correctly
2022-07-12 15:09:55,143 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '7f1a65b8a924e6c050221c52337f2efe', 'namespace': 'Appliance.System.Online', 'timestamp': 1657602595, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'f78c0591f10467fd35e896da1ebb2f29', 'from': '/appliance/2104290289693390850448e1e96f34f2/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:10:34,247 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '396220e5055b6c3076703062344b7c64', 'namespace': 'Appliance.System.Online', 'timestamp': 1657602634, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '361d0e19eaf6f5b8e46ea3f88ae0afe2', 'from': '/appliance/2104290289693390850448e1e96f34f2/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:05,843 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': 'ac114de0bcb45a3038946bb9ce8bc3ca', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604645, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'b04436a57f52d740bcfcfa14f38d3cde', 'from': '/appliance/2103253652925190847048e1e96694a6/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:05,844 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': 'f9700fc99ffdf57ed02f2af37592480e', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604645, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'cd8fbf1fd91f7937530aa06fdd8ed55d', 'from': '/appliance/21020580286306290d4248e1e94aec87/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:10,642 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '4989d2007f4a9a264442b3a2d42dc80f', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604650, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '660dc94d45774d3459fb68431a867d31', 'from': '/appliance/21020510538920290d4248e1e94ae995/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:11,044 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '83737617cb6982713c1c9d22c5a4d95d', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604650, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'a770dd33f8291aad2c0fb84d42099ac9', 'from': '/appliance/21020580286306290d4248e1e94aec87/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:11,642 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '290fc882499ddbd735f77254674e1bb5', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604651, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '81f4e1fdcd353ee982b13883acb1d7e1', 'from': '/appliance/21020510538920290d4248e1e94ae995/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:11,743 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '61fbc42ba0c7369ef3319959c2b09fa4', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604651, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'e4050e51c42ba19da5ce63b210d38a62', 'from': '/appliance/2103253413046290847048e1e966a5c2/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:11,843 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 690, 'timestamp': 1657604651, 'sign': '792a6c17d8e3e64877f28305a1a46362', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': '81dea8bc5ae370bbe03414c1d0b82df6', 'from': '/appliance/21020580286306290d4248e1e94aec87/publish'}, 'payload': {'light': {'transform': -1, 'temperature': 100, 'rgb': 16753920, 'luminance': 30, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:12,041 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '311a6afd455aa0015eae5bbd39f742c1', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604651, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '72a5eb2194ea47da93704db7aa10fe43', 'from': '/appliance/21020568578299290d4248e1e94b0298/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:12,043 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': 'cef5e46ebcd4df8c8d2e3f0ab90dd5c1', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604651, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'aeb11fa47556c444dbf0907505042c4e', 'from': '/appliance/2103253903825090847048e1e966ac3e/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:12,142 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '49cb8863f7c5229b4636fed808705c54', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604652, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '211edec3e81133a8c9445e7f1cd77147', 'from': '/appliance/21020506709202290d4248e1e94ad9b1/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:12,144 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '8b412b108446afa70565deaf3a8cfadd', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604652, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'a1977d500c66a50cca5a61bf8692928e', 'from': '/appliance/21020510901628290d4248e1e94ae19d/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:12,182 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 224, 'timestamp': 1657604652, 'sign': '7699a59bfb34afd35cff6d5a18e4ad20', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': '59d8a770322ba7f3ecfb07e90182c046', 'from': '/appliance/21020510538920290d4248e1e94ae995/publish'}, 'payload': {'light': {'transform': -1, 'temperature': 100, 'rgb': 16753920, 'luminance': 30, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:12,643 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': 'a59eed0815fb15891990d945e7f290e6', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604652, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '3fff666c71aa5b83df04eaead49879a5', 'from': '/appliance/2103250510838390847048e1e966aea4/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:12,742 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '9c87d1dc405c29cf8f551ef662a37d4b', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604652, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '19d0fa6450ce4dc5a93bba2637d750d6', 'from': '/appliance/2103253652925190847048e1e96694a6/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:12,870 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 722, 'timestamp': 1657604652, 'sign': 'dbd6d38443f20c5b37375d18cf8741fa', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': '69131abdf55bc3d747ccb96033bf9c26', 'from': '/appliance/21020510901628290d4248e1e94ae19d/publish'}, 'payload': {'light': {'transform': -1, 'temperature': 100, 'rgb': 16753920, 'luminance': 30, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:12,883 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 757, 'timestamp': 1657604652, 'sign': '12b9e29627785e885d088d0253141ac1', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': '6bda9181e0e1b496301f1ecdaf10cee9', 'from': '/appliance/2103253903825090847048e1e966ac3e/publish'}, 'payload': {'light': {'transform': -1, 'temperature': 100, 'rgb': 16753920, 'luminance': 30, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:12,943 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '1f502b1f572ca5403bb72cdf0dfb6b5f', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604652, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '26b56b18a68076e00d3f852d9f36f885', 'from': '/appliance/2103250510838390847048e1e966aea4/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:13,244 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '9ac4db00dc6cc9a8ad4ac7ae0e6123c3', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604653, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '67515b488e0b04f4a3e92e6cb2370a76', 'from': '/appliance/2103253413046290847048e1e966a5c2/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:13,288 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 235, 'timestamp': 1657604653, 'sign': '80210221612e1d8e68bd59b62374087a', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': '45197930a8e092d174074b24237bafd2', 'from': '/appliance/2103253652925190847048e1e96694a6/publish'}, 'payload': {'light': {'transform': -1, 'temperature': 100, 'rgb': 16753920, 'luminance': 30, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:13,443 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': 'd0a78ad6a3927deb1ccdc943c9c97228', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604653, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': 'e35d7c214243cfdd50b0d21d662298ec', 'from': '/appliance/21020594730951290d4248e1e94ae23a/publish'}, 'payload': {'online': {'status': 2}}}
2022-07-12 15:44:13,754 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 727, 'timestamp': 1657604653, 'sign': '4320601f3bc831e437d80c3e16bf2989', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': 'eee33141874599838ef99829250b94a8', 'from': '/appliance/2103253413046290847048e1e966a5c2/publish'}, 'payload': {'light': {'transform': 0, 'temperature': 100, 'rgb': 16753920, 'luminance': 100, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:13,941 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '329488d2d687cb6fb0f8e93ecc1842d5', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604653, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '0f93b8b1116f630551409f5f22012ef8', 'from': '/appliance/21020506709202290d4248e1e94ad9b1/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:14,849 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 763, 'timestamp': 1657604654, 'sign': '77081022608793c080be0c28178df3a0', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': 'ea7522e2e94d35ec20d08455678330fc', 'from': '/appliance/21020506709202290d4248e1e94ad9b1/publish'}, 'payload': {'light': {'transform': 0, 'temperature': 100, 'rgb': 16753920, 'luminance': 100, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:18,341 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': 'c2f323ef2968b29e0ae8329d422092ad', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604658, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '7c2c669df797db72eaae6e0436062fe0', 'from': '/appliance/21020594730951290d4248e1e94ae23a/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:18,875 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 28, 'timestamp': 1657604658, 'sign': '059f89ef5b21637e672ce079d848db33', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': 'f278adb8e03fd97623afbd8c4bcd0732', 'from': '/appliance/21020594730951290d4248e1e94ae23a/publish'}, 'payload': {'light': {'transform': -1, 'temperature': 100, 'rgb': 16753920, 'luminance': 100, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:31,244 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '11a496b22ec799cff8a4963180b8e5d5', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604671, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '9b3c3fdcb69d55448557cbdb9e6a6e39', 'from': '/appliance/21020568578299290d4248e1e94b0298/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:31,688 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 204, 'timestamp': 1657604672, 'sign': 'c7646b9e511b30d51293d2e68f278d23', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': '582ccc9faf53ef0e9d5fc23582715c30', 'from': '/appliance/21020568578299290d4248e1e94b0298/publish'}, 'payload': {'light': {'transform': -1, 'temperature': 100, 'rgb': 16753920, 'luminance': 30, 'channel': 0, 'capacity': 6}}}
2022-07-12 15:44:32,742 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'messageId': '2e050a188e0ab20a8cf4b4e1bfd038d3', 'namespace': 'Appliance.System.Online', 'timestamp': 1657604672, 'triggerSrc': 'CloudOnline', 'payloadVersion': 1, 'method': 'PUSH', 'sign': '5b3ff48f4b21b878b93c37ebe68e7f4f', 'from': '/appliance/2103250510838390847048e1e966aea4/publish'}, 'payload': {'online': {'status': 1}}}
2022-07-12 15:44:33,090 - Sniffer - INFO - USER-TOPIC (/app/1583979/subscribe) <- {'header': {'timestampMs': 27, 'timestamp': 1657604672, 'sign': '6f54ed39e9c4a76f6798410bad76df36', 'payloadVersion': 1, 'namespace': 'Appliance.Control.Light', 'method': 'PUSH', 'messageId': '4ff128c8584739c3623c3c31aedc9eda', 'from': '/appliance/2103250510838390847048e1e966aea4/publish'}, 'payload': {'light': {'transform': 0, 'temperature': 100, 'rgb': 16753920, 'luminance': 100, 'channel': 0, 'capacity': 6}}}
```

**Comment by albertogeniola** [comment]
Hi @chaayac ,
Let me understand: do you face the flickering problem with the brand new bulbs with HA off?

**Comment by advicepyro** [comment]
I'm getting this issue too, but I'm not sure if it's related to HA.

I got a pair of MSL120 bulbs and paired them right away to HA using this integration and the other Meross LAN integration. After a while, the bulbs would flicker slightly - it's like if the brightness went down by a few notches and then back up. I factory reset one of them (got it into the factory mode red-blue-green-whites mode, then out of it), deleted the Google Assistant integration in the Meross app, and added the bulb back in, but it still flickers.

**Comment by chaayac** [comment]
>Let me understand: do you face the flickering problem with the brand new bulbs with HA off?
I still faced the issue when I connected the bulbs to the same network that I initially configured them to connect to. Strangely, even new bulbs that I had never configured with HA started having this problem when connected to the same network.

I solved this by switching on my guest network and using that instead. I also created a new Meross account and moved everything onto there. Unfortunately I don't know which one of these 2 things actually resolved the issue, I didn't test them in isolation.
